src/App/Json.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TypeApplications      #-}
+
+module App.Json where
+
+import Control.Lens
+import Data.Aeson                as J
+import Data.Generics.Product.Any
+import Data.Monoid
+
+import qualified Antiope.Contract.SQS.FileChangeMessage as SQSM
+import qualified Antiope.Contract.SQS.ResourceChanged   as SQSM
+
+fileChangeMessageToResourceChanged :: J.Value -> J.Value
+fileChangeMessageToResourceChanged v = case J.fromJSON v of
+  J.Success (fcm :: SQSM.FileChangeMessage) -> J.toJSON $ SQSM.ResourceChanged
+    { SQSM.eventTime  = fcm ^. the @"eventTime"
+    , SQSM.uri        = "s3://" <> (fcm ^. the @"bucketName") <> "/" <> (fcm ^. the @"objectKey")
+    }
+  J.Error _                                 -> v
```

Comment:
How do we detect that this happened?

Reply:
In the case where the message can't be converted, the original is retained.

Reply:
So who's doing the checking?